### DISCUSSION TITLE
Does it make sense to create a `valgrind` CI job for `wgpu`?

### DISCUSSION THREAD

**Discussion by Luni-4** [gfx-rs/wgpu#4714] [Q&A] [answered]
I do not know if this idea makes sense, because reading around it seems that `valgrind` can produce a certain amount of false-positives for GPU-wise contexts. I'm asking to you, who are more experts in the field, in order to know if a new `GitHub` action in that direction could be useful. Thank you!

**Comment by cwfitzgerald** [comment, marked as answer]
I wouldn't expect false positives - I can run the test suite under valgrind fine. The main problem is that valgrind is _slow_ and running the whole test suite can take many minutes on a 10 core, let alone a 2 core CI machine.

**Reply by cwfitzgerald** [reply]
In short - running locally makes sense, in CI, not so much.

**Comment by Luni-4** [comment]
> I wouldn't expect false positives - I can run the test suite under valgrind fine. The main problem is that valgrind is _slow_ and running the whole test suite can take many minutes on a 10 core, let alone a 2 core CI machine.

This is true, but one can run `valgrind` either on specific tests or on the entire test suite before releasing a new version. Do you think that could be a good alternative?

**Reply by cwfitzgerald** [reply]
We don't really have enough bugs that valgrind to solve to bother with the maintenance of such a thing.
